src/renderer/components/icon/icon.tsx

```diff
@@ -15,22 +15,69 @@ import { withTooltip } from "../tooltip";
 import isNumber from "lodash/isNumber";
 import { decode } from "../../../common/utils/base64";
 
-export interface IconProps extends React.HTMLAttributes<any>, TooltipDecoratorProps {
-  material?: string;          // material-icon, see available names at https://material.io/icons/
-  svg?: string;               // svg-filename without extension in current folder
-  link?: LocationDescriptor;   // render icon as NavLink from react-router-dom
-  href?: string;              // render icon as hyperlink
-  size?: string | number;     // icon-size
-  small?: boolean;            // pre-defined icon-size
-  smallest?: boolean;            // pre-defined icon-size
-  big?: boolean;              // pre-defined icon-size
-  active?: boolean;           // apply active-state styles
-  interactive?: boolean;      // indicates that icon is interactive and highlight it on focus/hover
-  focusable?: boolean;        // allow focus to the icon + show .active styles (default: "true", when icon is interactive)
+export interface BaseIconProps {
```

Comment:
I separated out these props so that I could export this subset as part of the `KubeObjectHandlerRegistration` type.

The bonus of using doc comments it that our API docs should be better documented now.